cupy/core/fusion.py

```diff
@@ -649,7 +650,15 @@ def fuse(input_num=None, reduce=None, post_map=lambda x: x):
             If not assigned, post_map step is skipped.
     """
     util.experimental('cupy.core.fusion')
-    return lambda f: Fusion(f, input_num, reduce, post_map)
+
+    def wrapper(f, input_num=None, reduce=None, post_map=lambda x: x):
+        return Fusion(f, input_num, reduce, post_map)
+
+    if len(args) == 1 and len(kwargs) == 0 and callable(args[0]):
+        return functools.update_wrapper(wrapper(args[0]), args[0])
```

Comment:
I think `functools.wraps` cannot be used here as it is a decorator, and what we want to look like `args[0]` is a `Fusion` instance, not `wrapper`.